splunk-devops/src/main/groovy/com/splunk/splunkjenkins/UserActionDSL.groovy

```diff
@@ -36,25 +37,33 @@ public class UserActionDSL {
                         env: buildParameters, listener: listener);
                 Binding binding = new Binding();
                 binding.setVariable("splunkins", delegate);
-                ClassLoader cl = Jenkins.getActiveInstance().getPluginManager().uberClassLoader;
+                binding.setVariable("build", build);
+                binding.setVariable("env", buildParameters);
                 // the script written before SecureGroovyScript was introduced, and can not be migrated automatically
                 boolean legacyScript = SplunkJenkinsInstallation.get().isLegacyMode();
                 try {
+                    ApprovalContext context = ApprovalContext.create().withCurrentUser().withItemAsKey(build.getParent());
+                    //saving pending for approval
+                    String groovyScriptText = ScriptApproval.get().configuring(scriptText, GroovyLanguage.get(), context);
+                    //check approval, will throw UnapprovedUsageException
+                    ScriptApproval.get().using(groovyScriptText, GroovyLanguage.get());
```

Comment:
about "this mode can only work where there is a fixed script associated with a Job and configured by a clearly identified person", I think whole script mode is just store a hash in scriptApproval.xml which doesn't bind to Job or Person. When the hashes is not found, just store as PendingScript which doesn't use the context(Job or Person) for hashcode/equal method either, (equal is checked on script and lang only)